include/structured/Dialect/Substrait/IR/SubstraitOps.td

```diff
@@ -128,14 +128,79 @@ def Substrait_PlanRelOp : Substrait_Op<"relation", [
 }
 
 def Substrait_YieldOp : Substrait_Op<"yield", [
-    Terminator, HasParent<"::mlir::substrait::PlanRelOp">
+    Terminator,
+    ParentOneOf<[
+      "::mlir::substrait::PlanRelOp",
+      "::mlir::substrait::FilterOp"
```

Comment:
Order?

Reply:
(Already raised and addressed in #812, sorry for the double review effort.)